src/core/agent/services/signifyNotificationService.ts

```diff
@@ -0,0 +1,68 @@
+import { AgentService } from "./agentService";
+import { GenericRecordType, KeriNotification } from "../agent.types";
+import { Notification } from "./credentialService.types";
+// TODO: DUPLICATE WITH NOTIFICATION IN CRED SERVICE, HANDLE THIS
+export class SignifyNotificationService extends AgentService {
+  async onNotificationKeriStateChanged(
+    callback: (event: KeriNotification) => void
+  ) {
+    // eslint-disable-next-line no-constant-condition
+    while (true) {
+      const notifications = await this.agent.modules.signify.getNotifications();
+      for (const notif of notifications.notes) {
+        await this.processNotification(notif, callback);
+      }
+      await new Promise((rs) => {
+        setTimeout(() => {
+          rs(true);
+        }, 20000);
+      });
+    }
+  }
+
+  async processNotification(
+    notif: Notification,
+    callback: (event: KeriNotification) => void
+  ) {
+    const keriNoti = {
+      id: notif.i,
+      a: notif.a,
+      createdAt: new Date(),
+    } as KeriNotification;
+    // todo: save data
```

Comment:
Do we just need to store while waiting for multisig to complete?

Reply:
There is a case when the user received a join multisig request but they close the app before accepting the request. We save those notifications to handle them later.

Reply:
I thought we were able to mark notifications as read? So could we read the notification, join the multi-sig and then mark it as read?

Or does it take too long and you end up re-processing the notification?

I see this is just a loop fetching notifications - would be nice if in the future we can improve this to be more like the mediator where it fetches on start-up (pickup) and then streams in new ones, but I appreciate this isn't the case with Signify.

If we can't do this, then we can store but it would be nice to just store the multi-sig ones, and to delete them after they are done.

Reply:
Yeah, I am following the flow like how we are handling the credentials. The method we used to fetch the notifications does not support filters `this.signifyClient.notifications().list()` so it will be pretty inconvenient to get unhandled notifications. I think we should save them for now, and remove this save logic after Signify supports more filter options.

Reply:
When we fetch them into our wallet, can we not filter then if they are read or not?

Reply:
`this.signifyClient.notifications().list()` can only get 24 notifications in a call. So I think getting the notifications from the metadata would be better at the moment.

Reply:
OK, cool, sounds good then! :)